src/Our.Umbraco.StackedContent/Web/UI/App_Plugins/StackedContent/js/stackedcontent.controllers.js

```diff
@@ -55,34 +55,35 @@
             setDirty();
         };
 
-        $scope.copyToLocalStorage = function (evt, idx) {
-            var stackedContentItem = JSON.parse(JSON.stringify($scope.model.value[idx]));
-            stackedContentItem.key = "";
-            delete stackedContentItem.$$hashKey;
-
-            if (validateModel(stackedContentItem)) {
-                window.localStorage.setItem("StackedContentCopy", JSON.stringify(stackedContentItem));
-                notificationsService.success("Stacked Content", "Copied to clipboard.");
-                return;
+        $scope.copyContent = function (evt, idx) {
+            // TODO: Move this to InnerContent Service
+            var item = $scope.model.value[idx];
+            if (item && item.icContentTypeGuid) {
+                localStorageService.set("icContentJson", JSON.stringify(item, function (k, v) {
+                    if (k === "key" || k === "$$hashKey") {
+                        return undefined;
+                    }
+                    return v;
+                }));
+                allowPaste = true;
```

Comment:
I'm not convinced the allowPaste local variable is fool proof. Is there as possibility this could be false/true but there is something pastable int he clipboard? How expensive is the pastedAllowed() call?

Reply:
It felt pretty expensive. I'd added a `console.log` in `canPaste` - I had 5 StackedContent fields on a page, (with at least 10 blocks between them) - meaning it was being called for each field, then each row.

It felt overkill to get the data from localStorage, parse it and loop over the contentType GUIDs each time.

There's mostly a clever way to handle this?